packages/core/src/codewhisperer/commands/basicCommands.ts

```diff
@@ -424,64 +424,34 @@ export const signoutCodeWhisperer = Commands.declare(
 
 let _toolkitApi: any = undefined
 
-const registerToolkitApiCallbackOnce = once(async () => {
+const registerToolkitApiCallbackOnce = once(() => {
     getLogger().info(`toolkitApi: Registering callbacks of toolkit api`)
     const auth = Auth.instance
 
     auth.onDidChangeConnectionState(async e => {
-        // when changing connection state in Q, also change connection state in toolkit
-        if (_toolkitApi && 'setConnection' in _toolkitApi) {
+        if (_toolkitApi && 'declareConnection' in _toolkitApi) {
             const id = e.id
             const conn = await auth.getConnection({ id })
-            if (conn && conn.type === 'sso') {
-                getLogger().info(`toolkitApi: set connection ${id}`)
-                await _toolkitApi.setConnection({
-                    type: conn.type,
-                    ssoRegion: conn.ssoRegion,
-                    scopes: conn.scopes,
-                    startUrl: conn.startUrl,
-                    state: e.state,
-                    id: id,
-                    label: conn.label,
-                } as AwsConnection)
+            if (conn?.type === 'sso') {
+                getLogger().info(`toolkitApi: declare connection ${id}`)
+                _toolkitApi.declareConnection(
+                    {
+                        type: conn.type,
+                        ssoRegion: conn.ssoRegion,
+                        startUrl: conn.startUrl,
+                        id: id,
```

Comment:
Do we need the `id` of a connection still? Did this make sense when we shared connections, but isn't required anymore

Reply:
The delete connection callback only supplied id, so that was the main hash. I also didn't want to create conflicts with starturls of varying regions. But you can't have the same starturl in multiple regions, and most users will not have multiple sso logins to choose from anyway.

So instead I updated the delete connection callback to also return the stored profile so we can identify and hash on the start url.